src/processor_arm.cpp

```diff
@@ -662,13 +674,48 @@ static constexpr CPUSpec<CPU, feature_sz> cpus[] = {
     {"exynos-m2", CPU::samsung_exynos_m2, CPU::generic, UINT32_MAX, Feature::samsung_exynos_m2},
     {"exynos-m3", CPU::samsung_exynos_m3, CPU::generic, 0, Feature::samsung_exynos_m3},
     {"exynos-m4", CPU::samsung_exynos_m4, CPU::generic, 0, Feature::samsung_exynos_m4},
-    {"exynos-m5", CPU::samsung_exynos_m5, CPU::samsung_exynos_m4, 110000,
-     Feature::samsung_exynos_m5},
+    {"exynos-m5", CPU::samsung_exynos_m5, CPU::samsung_exynos_m4, 110000, Feature::samsung_exynos_m5},
     {"apple-a7", CPU::apple_a7, CPU::generic, 0, Feature::apple_a7},
 };
 #endif
 static constexpr size_t ncpu_names = sizeof(cpus) / sizeof(cpus[0]);
 
+static inline const CPUSpec<CPU,feature_sz> *find_cpu(uint32_t cpu)
+{
+    return ::find_cpu(cpu, cpus, ncpu_names);
+}
+
+static inline const CPUSpec<CPU,feature_sz> *find_cpu(llvm::StringRef name)
+{
+    return ::find_cpu(name, cpus, ncpu_names);
+}
+
+static inline const char *find_cpu_name(uint32_t cpu)
+{
+    return ::find_cpu_name(cpu, cpus, ncpu_names);
+}
+
+#if defined _CPU_AARCH64_ && defined _OS_DARWIN_
+static CPU get_cpu_name()
+{
+    char buffer[128];
+    size_t bufferlen = 128;
+    sysctlbyname("machdep.cpu.brand_string",&buffer,&bufferlen,NULL,0);
+    if(strcmp(buffer,"Apple M1") == 0)
+        return CPU::apple_m1;
```

Comment:
And what I was writing previously would just corresponds to `return {CPU::apple_m1, Feature::apple_m1};` in this branch and returning a different pair if a different cpu is detected. Going through `find_cpu` is also fine, just a bit unnecessary.....

Reply:
I agree, I think it's cleaner now